libtfhe/core/arithmetic/torus_utils_big.h

```diff
@@ -70,8 +70,9 @@ class TorusUtils<BigTorus> {
      * @return double value of the infinity norm
      */
     static double normInftyDist(const TORUS *t1, const TORUS *t2, const ZModuleParams<TORUS> *params) {
-        abort(); //TODO the code below won't do mod 1 properly (e.g. dist(0.01,0.99) is wrong): add a distance function to the torus
-        return abs(TorusUtils<TORUS>::to_double(t1, params)-TorusUtils<TORUS>::to_double(t2, params)); // quick and dirty :P
+        double diff = std::abs(TorusUtils<TORUS>::to_double(t1, params)-TorusUtils<TORUS>::to_double(t2, params));
+        diff = (diff > 0.5) ? 1.0 - diff : diff;
```

Comment:
This version of the distance is not precise, maybe we can put normInftyDist in the test lib?
(or add an allocator argument to compute t1-t2?)

Reply:
the exponent matters: for very small distances torus values, bigtorus to_double should return a tiny double with the right exponent (as small as 2^-1000).
Alternatively, if distance is only used for unit testing (i.e. good answer or random), then the function can be moved to testlib, and does not need to be accurate.

Reply:
You're right about the precision.
This function is used in TorusPolynomial to compute infinity norm, don't know if torus polynomial norm will be used in future.

Reply:
You're right about the precision.
This function is used in TorusPolynomial to compute infinity norm, don't know if torus polynomial norm will be used in future.